gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposeCompilerKotlinSupportPlugin.kt

```diff
@@ -8,11 +8,53 @@ package org.jetbrains.compose
 import org.gradle.api.Project
 import org.gradle.api.provider.Provider
 import org.jetbrains.compose.internal.ComposeCompilerArtifactProvider
+import org.jetbrains.compose.internal.KOTLIN_JVM_PLUGIN_ID
+import org.jetbrains.compose.internal.KOTLIN_MPP_PLUGIN_ID
+import org.jetbrains.compose.internal.Version
 import org.jetbrains.compose.internal.mppExtOrNull
 import org.jetbrains.compose.internal.service.ConfigurationProblemReporterService
 import org.jetbrains.compose.internal.webExt
-import org.jetbrains.kotlin.gradle.plugin.*
+import org.jetbrains.kotlin.gradle.plugin.KotlinBasePlugin
+import org.jetbrains.kotlin.gradle.plugin.KotlinCompilation
+import org.jetbrains.kotlin.gradle.plugin.KotlinCompilerPluginSupportPlugin
+import org.jetbrains.kotlin.gradle.plugin.KotlinPlatformType
+import org.jetbrains.kotlin.gradle.plugin.KotlinTarget
+import org.jetbrains.kotlin.gradle.plugin.SubpluginArtifact
+import org.jetbrains.kotlin.gradle.plugin.SubpluginOption
+import org.jetbrains.kotlin.gradle.plugin.getKotlinPluginVersion
 import org.jetbrains.kotlin.gradle.targets.js.ir.KotlinJsIrTarget
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+internal fun Project.configureComposeCompilerPlugin() {
+    plugins.withId(KOTLIN_MPP_PLUGIN_ID) { plugin ->
+        configureComposeCompilerPlugin(plugin as KotlinBasePlugin)
+    }
+    plugins.withId(KOTLIN_JVM_PLUGIN_ID) { plugin ->
+        configureComposeCompilerPlugin(plugin as KotlinBasePlugin)
+    }
+}
+
+internal const val newCompilerIsAvailableVersion = "2.0.0-RC1"
```

Comment:
Some thoughts:
- RC1 is the only version we can test this chnage at the moment.
- RC1 is okay for android+jvm projects where we can test this change. 

But we should not force 1.6.10-beta01 users to apply a new compose-compiler-gradle plugin, because it doesn't support compose multiplatform yet.

Perhaps we don't merge this now, so it won't go into 1.6.10-beta01. 
Before merging, we can update to RC2 (hopefully) or RC3 version here.